infra/scripts/deploy/src/main.rs

```diff
@@ -0,0 +1,124 @@
+    run(Cli::Check {
+        network: "testnet".to_string(),
+        mpc_contract_id: "v5.multichain-mpc-dev.testnet".parse().unwrap(),
+    })
+    .unwrap();
+}
+
+fn run(cli: Cli) -> anyhow::Result<()> {
+    match cli {
+        Cli::Check {
+            network,
+            mpc_contract_id,
+        } => {
+            let rt = tokio::runtime::Builder::new_multi_thread()
+                .enable_all()
+                .build()?;
+            rt.block_on(async {
+                let worker = near_workspaces::sandbox().await.unwrap();
+                let local = worker.dev_create_account().await.unwrap();
+                let importer =
+                    match network.as_str() {
+                        "testnet" => worker
+                            .import_contract(&mpc_contract_id, &near_workspaces::testnet().await?),
+                        "mainnet" => worker
+                            .import_contract(&mpc_contract_id, &near_workspaces::mainnet().await?),
+                        _ => anyhow::bail!("Unknown network: {network}"),
+                    };
+
+                let contract = importer
+                    .with_data()
+                    .dest_account_id(local.id())
+                    .transact()
+                    .await?;
+
+                let execution = contract.as_account().deploy(CONTRACT_BYTES).await?;
+                let new_contract = execution.into_result()?;
+
+                // Check the contract state is migratable:
+                let migrate = new_contract
+                    .call("migrate")
+                    .max_gas()
+                    .transact()
+                    .await
+                    .unwrap();
+                dbg!(&migrate);
+
+                // Check the contract state is viewable:
+                let view = new_contract.view("state").await.unwrap();
+                dbg!(&view);
+                let state: mpc_contract::ProtocolContractState = view.json()?;
+                println!("Contract state: {state:#?}");
+
+                let msg = "hello world";
+                let path = "deploy-check";
+                let (_, _, payload_hash) = process_message(msg).await;
+                let request = mpc_contract::primitives::SignRequest {
+                    payload: payload_hash,
+                    path: path.into(),
+                    key_version: 0,
+                };
+
+                // Check that we can call into `sign`:
+                let status = contract
+                    .call("sign")
+                    .args_json(serde_json::json!({
+                        "request": request,
+                    }))
+                    .deposit(near_workspaces::types::NearToken::from_yoctonear(1))
+                    .max_gas()
+                    .transact_async()
+                    .await?;
+
+                let status = status.await?;
+                dbg!(&status);
+                assert!(status.is_failure(), "expected to timeout");
```

Comment:
Is it possible fail due to deserialize certain state failed (caused by migration)? Should we check execution outcome fail reason